src/pages/index.js

```diff
@@ -0,0 +1,372 @@
+      cardsList.append(cardElement);
+    });
+  })
+  .catch(console.error);
+
+function handleLike(evt, id, cardLikedBtn){
+  const isLiked = cardLikedBtn.classList.contains('card__like-button_liked');
+  
+  api. handleLikeStatus(id, isLiked) 
+  .then(() => {
+    cardLikedBtn.classList.toggle("card__like-button_liked");
+  })
+  .catch(console.error);
+}
+
+previewModalcloseBtn.addEventListener("click", () => {
+  closeModal(previewModal);
+});
+
+let selectedCard, selectedCardId;
+
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+
+  const cardNameEl = cardElement.querySelector(".card__title");
+  const cardImageEl = cardElement.querySelector(".card__image");
+  const cardLikedBtn = cardElement.querySelector(".card__like-button");
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-button");
+
+  cardNameEl.textContent = data.name;
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  
+  if (data.isLiked) { 
+    cardLikedBtn.classList.add("card__like-button_liked");
+  }
+
+  cardLikedBtn.addEventListener("click", (evt) => 
+    handleLike(evt, data._id, cardLikedBtn));
+
+  cardImageEl.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalCaptionEl.textContent = data.name;
+    previewModalImageEl.src = data.link;
+    previewModalImageEl.alt = data.name;
+  });
+
+  cardDeleteBtn.addEventListener("click", () => 
+    handleDeleteCard(cardElement, data._id, cardDeleteBtn));
+
+  return cardElement;
+}
+
+function handleDeleteCard(cardElement, cardId, cardDeleteBtn) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+
+  setButtonText(avatarDeletetBtn, true, "Delete", "Deleting...");
+
+  api.deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(avatarDeletetBtn, false, "Delete");
+    });
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  const profileSubmitBtn = evt.submitter;
+
+  setButtonText(profileSubmitBtn, true, "Save", "Saving...");
+
+  const updatedUserInfo = { 
+    name: editModalNameInput.value,
+    about: editModalAboutInput.value 
+  };
+
+  api.editUserInfo(updatedUserInfo)  
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      editFormElement.reset();
+      closeModal(editModal);
+
+      api.getAppInfo()
+        .then(([cards, userInfo]) => {
+          profileName.textContent = userInfo.name;
+          profileDescription.textContent = userInfo.about;
+        })
+        .catch(console.error);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(profileSubmitBtn, false, "Save");
+    });
+}
+
+function toggleSubmitButton() {
```

Comment:
`toggleSubmitButton` is not needed because you have everything inside `validation.js`. You handle the submit button **universally** there